rhsm/rhsm-utils.c

```diff
@@ -181,6 +181,97 @@ rhsm_json_array_is_subset_of_hash_table (JsonArray  *array,
   return TRUE;
 }
 
+/*
+ * Join a baseurl (hostname) and url (full or relpath).
+ *
+ * If url is a full url, just return it. Otherwise combine
+ * it with base, skipping redundant seperators if needed.
+ *
+ * Simulate the behavior of subscription manager.
+ */
+static gchar *
+rhsm_url_base_join (const gchar *base,
+                    const gchar *url)
+{
+  /* handle special cases similar to subscription manager */
+  if (!url || *url == '\0')
+    return g_strdup ("");
+  if (strstr (url, "://"))
+    return g_strdup (url);
+  if (!base || *base == '\0')
+    return g_strdup (url);
+
+  /* parse URI, split to schema, host, and path */
+  g_autofree gchar *schema = NULL;
+  g_autofree gchar *host = NULL;
+  g_autofree gchar *path = NULL;
+  gchar *tmp = strstr (base, ":");
+  if (tmp)
+    schema = g_strndup (base, tmp - base);
+  if (schema)
```

Comment:
Please could you add a comment why you strip "://", but keeps ":/". Funny part is that curl is able to handle https:/google.com, but not DNF. Try `curl https:/download.copr.fedorainfracloud.org/results/jmracek/VMaaS/epel-8-x86_64/01422446-dnf-plugins-core/builder-live.log.gz --output some.file`